x-pack/plugins/code/server/routes/socket.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Server } from 'hapi';
+import { Socket } from 'socket.io';
+
+import { Log } from '../log';
+import { SocketService } from '../socket_service';
+
+export function socketRoute(server: Server, socketService: SocketService, log: Log) {
+  const socketIO = socketService.io;
+
+  socketIO.on('connection', (socket: Socket) => {
+    log.debug(`User ${socket.id} connected, attaching handlers and register socket.`);
+
+    // TODO(mengwei): apply the same security check as Canvas does.
```

Comment:
Already have a PR in here: https://github.com/elastic/kibana/pull/29088. Should be checked in in one day or two.

Reply:
Thanks for the PR!

> Also, just want to clarify a bit in here, even it's a socket.io connection in here, Code does not need bi-directional communication, but server pushing messages to the client only without accepting any messages/commands from the client. Do you think this could ease the security concern to some extent?

You'll still want to authenticate the connection. Otherwise, an unauthenticated or unauthorized user could establish a socket connection here, and start receiving information about Code's repositories (etc). They won't be able to issue commands since it's not bi-directional, but they'd still get access to information they shouldn't have.

It's not _quite_ as bad as an unauthenticated bi-directional channel, but it doesn't ease the security concern here. Thanks for working on a fix!